astrometry/include/PixelMap.h

```diff
@@ -332,5 +334,37 @@ namespace astrometry {
     void operator=(const ColorTerm& rhs) = delete;
   
   };
+
+  class ASTMap: public PixelMap {
+    // Class representing a constant shift of position
+  public:
+    ASTMap(const ast::Mapping& mapping_, std::string name_=""):
+      PixelMap(name_), mapping(mapping_) {}
+
+    virtual PixelMap* duplicate() const;
+
+    ~ASTMap() = default;
+    
+    static string type() {return "AST";}
+
+    ast::Mapping mapping;
+
+    void toWorld(double xpix, double ypix,
+                 double &xworld, double &yworld,
+                 double color=astrometry::NODATA) const;
+    void toPix(double xworld, double yworld,
+               double &xpix, double &ypix,
+               double color = astrometry::NODATA) const;
+
+#ifdef USE_YAML
+    // AST maps are not expected to be used for output, so we do not
+    // attempt to write a full description to yaml.
```

Comment:
We could, because AST mappings can be serialized to ASDF YAML.

Reply:
Good to know. I'm not going to implement that now, because I think it's pretty unlikely it will be needed.